Add bulk redirects to Pages how to section. (#3454)
cloudflare/cloudflare-docs: products/pages/src/content/how-to/use-bulk-redirects.md

---
pcx-content-type: how-to
---

# Handle redirects with Bulk Redirects

> In this tutorial, you will learn how to use Bulk redirects to handle redirects that are over `100` on your Pages projects. Since [`_redirects`](https://developers.cloudflare.com/pages/platform/limits#redirects) have a maximum of 100 redirects. 

Reviewer: > that are over `100`...

replace with

> if you need more than the 100 supported by the [`_redirects`] file.

![Bulk redirects option](./media/bulk_redirects_option.png)

This will take you to the bulk redirects UI where you will create a new Bulk Redirects list. 

![Create a new Bulk redirects list](./media/create_a_new_bulk_redirect_list.png)

Then create a new list, and in the content type select Redirect. This will lead you to add your redirect Source URL and Target URL, you can also specific the status code for each redirect. 

<Aside type="note">

You can also upload a CSV file of all your redirects, however your redirects can not be a relative path.

</Aside>

If for example you set the Source URL to `https://www.example.dev/examples` and the target to `https://www.example.dev/pages/examples` with a status code of `301`, whenever the Source is requested it will be permantly redirected to the Target.

After this, go back to the create Bulk Redirects button and select the list you just created. You would have to set a name for the **Page Rule** you are creating. 

![Create a new Bulk redirects](./media/create_new_bulk_redirect.png)

Finally, select **Save and deploy** and wait a few seconds for your rule to propagate. 

> 

Reviewer: Should we add a note here that these apply _in addition to_ and _in front of_ any redirects in your _redirects file?

Author: Yea, I will add a note as well.

---
pcx-content-type: how-to
---

# Handle redirects with Bulk Redirects

In this tutorial, you will learn how to use Bulk redirects to handle redirects if you need more than the 100 supported by the [`_redirects`](https://developers.cloudflare.com/pages/platform/limits#redirects) file. 

<Aside type="Note">
  The redirects defined in the _redirects file of your build folder can work together with your Bulk Redirects. In case of duplicates, Bulk redirects will run infront of your Pages project where your other redirects live.
</Aside>

> To do this, go to the [project overview dashboard](https://dash.cloudflare.com?to=/:account/:zone/rules) and under **Rules** select Bulk Redirects. 

Reviewer 2: ```suggestion
To do this, log in to the [Cloudflare dashboard](https://dash.cloudflare.com?to=/:account/:zone/rules) > **Rules** > **Bulk Redirects**.
```

![Bulk redirects option](./media/bulk_redirects_option.png)

> This will take you to the bulk redirects UI where you will create a new Bulk Redirects list. 

Reviewer 2: ```suggestion
In **Bulk Redirects**, select **Create a new Bulk Redirects list**.
```

![Create a new Bulk redirects list](./media/create_a_new_bulk_redirect_list.png)

> Then create a new list, and in the content type select Redirect. This will lead you to add your redirect Source URL and Target URL, you can also specific the status code for each redirect. 

Reviewer 2: ```suggestion
Create a new list, and in the content type, select **Redirect**. You will be prompted to add your redirect **Source URL** and **Target URL**. You can also specify the **Status** code for each redirect.
```

> <Aside type="note">
> 
> You can also upload a CSV file of all your redirects, however your redirects can not be a relative path.
> 
> </Aside>
> 
> If for example you set the Source URL to `https://www.example.dev/examples` and the target to `https://www.example.dev/pages/examples` with a status code of `301`, whenever the Source is requested it will be permantly redirected to the Target.
> 
> After this, go back to the create Bulk Redirects button and select the list you just created. You would have to set a name for the **Page Rule** you are creating. 

Reviewer 2: ```suggestion
<Aside type="note">

You can upload a CSV file of all your redirects. However, your redirects cannot be a relative path.

</Aside>

If you set the **Source URL** to `https://www.example.dev/examples` and the **Target URL** to `https://www.example.dev/pages/examples` with a status code of `301`, whenever the **Source URL** is requested, it will be permanently redirected to the **Target URL**.

After this, go back to **Bulk Redirects** > **Create Bulk Redirects**  > set a **Rule name** > and select the desired list.
```

---
pcx-content-type: how-to
---

# Handle redirects with Bulk Redirects

> In this tutorial, you will learn how to use Bulk redirects to handle redirects if you need more than the 100 supported by the [`_redirects`](https://developers.cloudflare.com/pages/platform/limits#redirects) file. 

Reviewer 2: ```suggestion
In this tutorial, you will learn how to use [Bulk Redirects (beta)](https://developers.cloudflare.com/rules/bulk-redirects) to handle redirects that surpasses the 100 redirect rules limit set by Pages. A [`_redirects`](https://developers.cloudflare.com/pages/platform/limits#redirects) file has a maximum of 100 redirect rules.
```

<Aside type="Note">

> The redirects defined in the _redirects file of your build folder can work together with your Bulk Redirects. In case of duplicates, Bulk redirects will run infront of your Pages project where your other redirects live.

Reviewer 2: ```suggestion
The redirects defined in the `_redirects` file of your build folder can work together with your Bulk Redirects. In case of duplicates, Bulk Redirects will run in front of your Pages project, where your other redirects live.
```

</Aside>

> To do this, log in to the [Cloudflare dashboard](https://dash.cloudflare.com?to=/:account/:zone/rules) > **Rules** > **Bulk Redirects**.

Reviewer 2: ```suggestion
To use Bulk Redirects, log in to the [Cloudflare dashboard](https://dash.cloudflare.com?to=/:account/:zone/rules) > **Rules** > **Bulk Redirects**.
```

![Bulk redirects option](./media/bulk_redirects_option.png)

In **Bulk Redirects**, select **Create a new Bulk Redirects list**.

![Create a new Bulk redirects list](./media/create_a_new_bulk_redirect_list.png)

Create a new list, and in the content type, select **Redirect**. You will be prompted to add your redirect **Source URL** and **Target URL**. You can also specify the **Status** code for each redirect.

<Aside type="note">

You can upload a CSV file of all your redirects. However, your redirects cannot be a relative path.

</Aside>

If you set the **Source URL** to `https://www.example.dev/examples` and the **Target URL** to `https://www.example.dev/pages/examples` with a status code of `301`, whenever the **Source URL** is requested, it will be permanently redirected to the **Target URL**.

After this, go back to **Bulk Redirects** > **Create Bulk Redirects**  > set a **Rule name** > and select the desired list.

![Create a new Bulk redirects](./media/create_new_bulk_redirect.png)

> Finally, select **Save and deploy** and wait a few seconds for your rule to propagate. 

Reviewer 3: We should probably also mention some of the options Bulk Redirects have. Mentioning that having like `example.com/` with "Subpath mathing" will match `example.com/*`
(I'm not actually sure on the specifics of how this works but it's a common thing people want so worth mentioning)
